index.ts

```diff
@@ -0,0 +1,85 @@
+interface ITree {
+  value?: number;
+  left?: ITree;
+  right?: ITree;
+}
+
+enum TreeDirection {
+  left = 'left',
+  right = 'right'
+}
+
+class Tree implements ITree {
+  value: number | undefined;
+  left: ITree;
+  right: ITree;
+  level: number;
```

Comment:
I think instead of so many classes we can just implemnt the tree on simple js object
```suggestion
root: ITree
```

Then to make it more readalbe let's change the name `ITree` to `ITreeNode`

Reply:
you suggested not to use class at all?